crates/server/src/store/in_memory/authority.rs

```diff
@@ -1615,11 +1615,18 @@ impl Authority for InMemoryAuthority {
                             _ => None,
                         };
 
-                        Ok(closest_encloser_match
-                            .into_iter()
-                            .chain(next_closer_name_cover)
-                            .chain(wildcard_record)
-                            .collect())
+                        let mut records = BTreeMap::new();
```

Comment:
I think a simpler way to do this would be to first collect into a `Vec`, then use `[T]::sort_by_key()` and then `Vec::dedup_by_key()`.

Reply:
Included this in a refactoring PR, #2885.